pilot/pkg/networking/core/v1alpha3/httproute.go

```diff
@@ -181,9 +180,27 @@ func (configgen *ConfigGeneratorImpl) buildSidecarOutboundHTTPRouteConfig(node *
 
 	util.SortVirtualHosts(virtualHosts)
 
-	if features.EnableFallthroughRoute.Get() && !useSniffing {
+	if !useSniffing {
 		// This needs to be the last virtual host, as routes are evaluated in order.
 		if util.IsAllowAnyOutbound(node) {
+			egressCluster := util.PassthroughCluster
+			// no need to check for nil value as the previous if check has checked
+			if node.SidecarScope.OutboundTrafficPolicy.EgressProxy != nil {
+				// user has provided an explicit destination for all the unknown traffic.
+				// build a cluster out of this destination
+				serviceFound := false
+				for _, service := range push.Services(node) {
```

Comment:
should we utilize something else than a linear lookup here? `ServiceForHostname`?

Reply:
Ah good point!

Reply:
We can make use of the existing `ServiceByHostnameAndNamespace`